ビルドが成功しない

https://twitter.com/cat_in_136/status/682420271983144960

[gulp-slim](https://www.npmjs.com/package/gulp-slim)の指示に従って、`gem install slim -v '>= 3.0.2'`した後

```
root@6414c1e7e945:/tmp/issue-maps# node_modules/.bin/gulp
[11:42:47] Requiring external module coffee-script/register
[11:42:47] Using gulpfile /tmp/issue-maps/gulpfile.coffee
[11:42:47] Starting 'build:ts'...
[11:42:48] Starting 'build:slim'...
[11:42:48] Starting 'build:sass'...
[11:42:48] Starting 'build:static'...
[11:42:48] Starting 'build:bower'...
[11:42:48] Finished 'build:bower' after 16 ms
[11:42:48] Finished 'build:static' after 127 ms
src/js/Application.ts(1,1): error TS6053: File '/tmp/issue-maps/typings/tsd.d.ts' not found.
src/js/Application.ts(5,14): error TS2304: Cannot find name 'angular'.
src/js/Application.ts(19,24): error TS2503: Cannot find namespace 'ng'.
src/js/Application.ts(23,21): error TS2304: Cannot find name 'angular'.
src/js/Application.ts(39,30): error TS2503: Cannot find namespace 'ng'.
src/js/Application.ts(43,31): error TS2503: Cannot find namespace 'ng'.
src/js/Application.ts(43,72): error TS2503: Cannot find namespace 'ng'.
src/js/Application.ts(71,22): error TS2307: Cannot find module './Setting'.
src/js/directives/IssueDirective.ts(1,1): error TS6053: File '/tmp/issue-maps/typings/tsd.d.ts' not found.
src/js/directives/IssueDirective.ts(4,48): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssueDirective.ts(30,32): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssueDirective.ts(30,58): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssueDirective.ts(30,100): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssuesListDirective.ts(1,1): error TS6053: File '/tmp/issue-maps/typings/tsd.d.ts' not found.
src/js/directives/IssuesListDirective.ts(4,53): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssuesListDirective.ts(29,32): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssuesListDirective.ts(29,59): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssuesListDirective.ts(29,93): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssuesListDirective.ts(29,135): error TS2503: Cannot find namespace 'ng'.
src/js/directives/IssuesListDirective.ts(29,181): error TS2503: Cannot find namespace 'ng'.
src/js/directives/MapsDirective.ts(1,1): error TS6053: File '/tmp/issue-maps/typings/tsd.d.ts' not found.
src/js/directives/MapsDirective.ts(3,47): error TS2503: Cannot find namespace 'ng'.
src/js/directives/MapsDirective.ts(77,32): error TS2503: Cannot find namespace 'ng'.
src/js/directives/MapsDirective.ts(77,62): error TS2503: Cannot find namespace 'ng'.
src/js/directives/MapsDirective.ts(77,99): error TS2503: Cannot find namespace 'ng'.
src/js/directives/MapsDirective.ts(77,137): error TS2503: Cannot find namespace 'ng'.
src/js/directives/RootDirective.ts(1,1): error TS6053: File '/tmp/issue-maps/typings/tsd.d.ts' not found.
src/js/directives/RootDirective.ts(5,47): error TS2503: Cannot find namespace 'ng'.
src/js/directives/RootDirective.ts(20,32): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(1,22): error TS2307: Cannot find module '../Setting'.
src/js/services/IssueService.ts(7,31): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(7,64): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(7,94): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(10,19): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(55,26): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(76,58): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(93,35): error TS2503: Cannot find namespace 'ng'.
src/js/services/IssueService.ts(97,9): error TS2304: Cannot find name 'angular'.
src/js/services/IssueService.ts(106,46): error TS2503: Cannot find namespace 'ng'.
[11:42:50] TypeScript: 39 semantic errors
[11:42:50] TypeScript: emit succeeded (with errors)

events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: Slim error in file (/tmp/issue-maps/src/html/index.slim):
Encoding::InvalidByteSequenceError: (__TEMPLATE__) is not valid US-ASCII
  Use --trace for backtrace.

root@6414c1e7e945:/tmp/issue-maps# npm list tsd  
npm info it worked if it ends with ok
npm info using npm@3.3.12
npm info using node@v5.3.0
issue-maps@1.0.0 /tmp/issue-maps
`-- (empty)
root@6414c1e7e945:/tmp/issue-maps#
```

tsdないからエラー起こるのは当たり前。

:+1: b46c082

Next stage...

```
root@6414c1e7e945:/tmp/issue-maps# node_modules/.bin/gulp      
[13:04:02] Requiring external module coffee-script/register
[13:04:02] Using gulpfile /tmp/issue-maps/gulpfile.coffee
[13:04:02] Starting 'build:tsd'...
[13:04:02] [gulp-tsd] execute command: reinstall
[13:04:02] [gulp-tsd] config: ./tsd.json
[13:04:02] Starting 'build:ts'...
[13:04:03] Starting 'build:slim'...
[13:04:03] Starting 'build:sass'...
[13:04:03] Starting 'build:static'...
[13:04:03] Starting 'build:bower'...
[13:04:03] Finished 'build:bower' after 17 ms
[13:04:03] [gulp-tsd] latest: undefined
[13:04:03] [gulp-tsd] running...
[13:04:03] Finished 'build:static' after 160 ms
src/js/Application.ts(71,22): error TS2307: Cannot find module './Setting'.
src/js/services/IssueService.ts(1,22): error TS2307: Cannot find module '../Setting'.
[13:04:06] TypeScript: 2 semantic errors
[13:04:06] TypeScript: emit succeeded (with errors)
Uncaught exception: { [Error: Slim error in file (/tmp/issue-maps/src/html/index.slim):
Encoding::InvalidByteSequenceError: (__TEMPLATE__) is not valid US-ASCII
  Use --trace for backtrace.
]
  message: 'Slim error in file (/tmp/issue-maps/src/html/index.slim):\nEncoding::InvalidByteSequenceError: (__TEMPLATE__) is not valid US-ASCII\n  Use --trace for backtrace.\n',
  showStack: false,
  showProperties: true,
  plugin: 'gulp-slim',
  __safety: { toString: [Function: bound ] } }
```

SlimのデフォルトエンコードがUS-ASCIIとかいう米国からの参入障壁（違

ていうか、下のようにして travis-ci で gulp build しないと永遠にたどり着かない気がする。

``` diff
diff --git a/gulpfile.coffee b/gulpfile.coffee
index e7b7646..397aee2 100644
--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -96,6 +96,11 @@ gulp.task "watch", ["build", "enable-watch-mode", "connect"], ->
   gulp.watch "./src/html/**/*.slim",    ["build:slim"]
   gulp.watch "./src/css/**/*.sass",    ["build:sass"]

+gulp.task "test", [
+  "build"
+  # TODO test
+]
+
 gulp.task "default", ["build"]

 gulp.task "clean", del.bind(null, ["lib/*", "public/*"])
diff --git a/package.json b/package.json
index aca8e9c..2d9e8c6 100644
--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"OK: npm install is OK\" && exit 0"
+    "test": "node_modules/.bin/gulp test"
   },
   "repository": {
     "type": "git",
```

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,8 @@ typings/
 .sass-cache/
 Setting.ts
 
+bower_components/
+
 .bundle
 Gemfile.local
 vendor

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,14 @@
+node_modules/
 aws.json
 lib/
 public/
+typings/
+.sass-cache/
 Setting.ts
+
+.bundle
+Gemfile.local
+vendor
+
+*~
+*.swp

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,8 +5,6 @@ node_js:
   - "4.2.4"
   - "5.3.0"
   - node
-before_install:
-  - gem update --system
-  - gem install sass
-  - gem install slim -v '>= 3.0.2'
 sudo: false
+before_install:
+ - bundle install --path vendor/bundle

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,4 +5,8 @@ node_js:
   - "4.2.4"
   - "5.3.0"
   - node
+before_install:
+  - gem update --system
+  - gem install sass
+  - gem install slim -v '>= 3.0.2'
 sudo: false

--- a/Gemfile
+++ b/Gemfile
@@ -1,0 +1,4 @@
+source "https://rubygems.org"
+
+gem "sass"
+gem "slim", ">= 3.0.2"

--- a/README.md
+++ b/README.md
@@ -1,4 +1,3 @@
-
 issue-maps
 ==========
 
@@ -8,8 +7,8 @@ AngularJSでGoogle Mapsを扱う。
 
 # Requires
 
-* ruby sass
-* ruby slim
+* ruby
+* bundler
 
 # Use
 
@@ -17,10 +16,7 @@ src/js/Setting.tsのredmineのUrlをSettingSample.tsを参考に書き換える�
 
 # Build
 ```
-$ gem install sass
-$ gem install slim -v '>= 3.0.2'
+$ bundle install --path vendor/bundle
 $ npm install
-$ bower install
-$ npm install -g gulp
 $ gulp build
 ```

--- a/README.md
+++ b/README.md
@@ -6,7 +6,21 @@ issue-maps
 
 AngularJSでGoogle Mapsを扱う。
 
-# requires
+# Requires
 
 * ruby sass
 * ruby slim
+
+# Use
+
+src/js/Setting.tsのredmineのUrlをSettingSample.tsを参考に書き換える．
+
+# Build
+```
+$ gem install sass
+$ gem install slim -v '>= 3.0.2'
+$ npm install
+$ bower install
+$ npm install -g gulp
+$ gulp build
+```

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -1,4 +1,5 @@
 gulp = require "gulp"
+bower = require "gulp-bower"
 $ = require("gulp-load-plugins")()
 del = require "del"
 fs = require "fs"
@@ -10,6 +11,9 @@ tsd = require "gulp-tsd"
 #watchify = require "watchify"
 watchify = require "gulp-watchify"
 source = require "vinyl-source-stream"
+
+gulp.task "bower:install", ->
+  bower()
 
 gulp.task "build", [
   "build:browserify"
@@ -62,7 +66,7 @@ gulp.task "build:sass", ->
   $.rubySass("src/css/**/*.sass", {bundleExec: true})
   .pipe gulp.dest("public/css")
 
-gulp.task "build:bower", ->
+gulp.task "build:bower", ["bower:install"], ->
   gulp.src "bower_components/**/*"
   .pipe gulp.dest "public/bower_components"
 

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -21,7 +21,6 @@ gulp.task "build", [
 gulp.task "build:pre", [
   #"build:coffee"
   "build:tsd"
-  "build:ts"
 ]
 
 gulp.task "build:coffee", ->
@@ -32,7 +31,7 @@ gulp.task "build:coffee", ->
 gulp.task "build:tsd", (callback) ->
   tsd {command: "reinstall", config: "./tsd.json"}, callback
 
-gulp.task "build:ts", ->
+gulp.task "build:ts", ["build:ts"], ->
   gulp.src "src/js/**/*.ts"
   .pipe $.typescript
     target:"es6"

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -55,11 +55,11 @@ gulp.task "build:browserify", ["build:pre"], watchify (watchify)->
 
 gulp.task "build:slim", ->
   gulp.src "src/html/**/*.slim"
-  .pipe $.slim()
+  .pipe $.slim({bundler: true})
   .pipe gulp.dest("public")
 
 gulp.task "build:sass", ->
-  $.rubySass("src/css/**/*.sass")
+  $.rubySass("src/css/**/*.sass", {bundleExec: true})
   .pipe gulp.dest("public/css")
 
 gulp.task "build:bower", ->

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -4,6 +4,8 @@ del = require "del"
 fs = require "fs"
 packageJson = require "./package.json"
 
+tsd = require "gulp-tsd"
+
 #browserify = require "browserify"
 #watchify = require "watchify"
 watchify = require "gulp-watchify"
@@ -18,6 +20,7 @@ gulp.task "build", [
 ]
 gulp.task "build:pre", [
   #"build:coffee"
+  "build:tsd"
   "build:ts"
 ]
 
@@ -25,6 +28,9 @@ gulp.task "build:coffee", ->
   gulp.src "src/js/**/*.coffee"
   .pipe $.coffee()
   .pipe gulp.dest("lib")
+
+gulp.task "build:tsd", (callback) ->
+  tsd {command: "reinstall", config: "./tsd.json"}, callback
 
 gulp.task "build:ts", ->
   gulp.src "src/js/**/*.ts"
@@ -95,6 +101,11 @@ gulp.task "watch", ["build", "enable-watch-mode", "connect"], ->
   gulp.watch "./src/js/**/*.ts",      ["build:ts"]
   gulp.watch "./src/html/**/*.slim",    ["build:slim"]
   gulp.watch "./src/css/**/*.sass",    ["build:sass"]
+
+gulp.task "test", [
+  "build"
+  # TODO test
+]
 
 gulp.task "default", ["build"]
 

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -20,7 +20,7 @@ gulp.task "build", [
 ]
 gulp.task "build:pre", [
   #"build:coffee"
-  "build:tsd"
+  "build:ts"
 ]
 
 gulp.task "build:coffee", ->
@@ -31,7 +31,7 @@ gulp.task "build:coffee", ->
 gulp.task "build:tsd", (callback) ->
   tsd {command: "reinstall", config: "./tsd.json"}, callback
 
-gulp.task "build:ts", ["build:ts"], ->
+gulp.task "build:ts", ["build:tsd"], ->
   gulp.src "src/js/**/*.ts"
   .pipe $.typescript
     target:"es6"

--- a/gulpfile.coffee
+++ b/gulpfile.coffee
@@ -105,6 +105,12 @@ gulp.task "watch", ["build", "enable-watch-mode", "connect"], ->
   gulp.watch "./src/html/**/*.slim",    ["build:slim"]
   gulp.watch "./src/css/**/*.sass",    ["build:sass"]
 
+gulp.task "serve", ["build", "enable-watch-mode"], ->
+  console.log $.connect
+  $.connect.server
+    root: "public"
+    host: "127.0.0.1"
+
 gulp.task "test", [
   "build"
   # TODO test

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"OK: npm install is OK\" && exit 0"
+    "test": "node_modules/.bin/gulp test"
   },
   "repository": {
     "type": "git",
@@ -29,6 +29,7 @@
     "gulp-ruby-sass": "^2.0.6",
     "gulp-s3": "^0.3.0",
     "gulp-slim": "0.3.0",
+    "gulp-tsd": "0.0.4",
     "gulp-typescript": "^2.9.1",
     "gulp-watchify": "git://github.com/marcello3d/gulp-watchify.git",
     "vinyl-source-stream": "^1.1.0",

--- a/package.json
+++ b/package.json
@@ -13,6 +13,7 @@
   "author": "diffshare",
   "license": "MIT",
   "devDependencies": {
+    "gulp-bower": "0.0.13",
     "babel-plugin-transform-runtime": "^6.3.13",
     "babel-preset-es2015": "^6.3.13",
     "babel-runtime": "^6.3.19",

--- a/src/js/Setting.ts
+++ b/src/js/Setting.ts
@@ -1,0 +1,6 @@
+export default {
+    endpoint_type: "redmine",
+    endpoint_url: "",
+    issues_url: "",
+    issue_url: ""
+};

